com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/textureset/TextureSetGenerator.java

```diff
@@ -261,23 +261,20 @@ public static SpriteGeometry buildConvexHull(BufferedImage image, SpriteTrimming
     private static SpriteGeometry.Builder createSpriteGeometryFromRect(Rect rect) {
         SpriteGeometry.Builder builder = SpriteGeometry.newBuilder();
 
-        int imageWidth = rect.getWidth();
-        int imageHeight = rect.getHeight();
         boolean rotated = rect.getRotated();
         builder.setRotated(rotated);
 
-        if (rotated)
-        {
-            // for legacy reasons, we need to rotate it back
-            // The geometry wants the size in unrotated form
-            builder.setWidth(imageHeight);
-            builder.setHeight(imageWidth);
-        }
-        else
-        {
-            builder.setWidth(imageWidth);
-            builder.setHeight(imageHeight);
-        }
+        // may be rotated
+        int imageWidth = rect.getWidth();
+        int imageHeight = rect.getHeight();
+
+        int originalImageWidth = rotated ? imageHeight : imageWidth;
+        int originalImageHeight = rotated ? imageWidth : imageHeight;
```

Comment:
The sprite geometry vertices are unit space, unrotated, so we need the original size.